hydrocarbon_problem/env/env.py

```diff
@@ -21,22 +25,34 @@
 
 class AspenDistillation(dm_env.Environment):
     """
-    Action space assumes unbounded continuous values. This environment has a very not-standard
-    step function (2 streams created from 1 stream - # TODO describe).
+    This environment has a very not-standard step function (2 streams created from 1 stream)
+    To manage this we have an observation object that contains "created states" which are the states
+    that have been recently produced by an action, and "upcoming state" which is the observation of
+    the next stream that will be used to create a distillation column.
+    We also split "done" and "discount" into values for the recently created streams
+    from a specific action - for indicating whether these streams are product streams that leave the process,
+    as well as an "overall" value which indicates whether the overall episode is complete
+    (this occurs when the maximum number of columns have been used, or if there are no non-product
+    streams yet to be acted on in the `._stream_numbers_yet_to_be_acted_on` object.
     """
-    flowsheet_api: BaseAspenDistillationAPI  # TODO: add actual API here
 
-    def __init__(self, initial_feed_spec: StreamSpecification = DEFAULT_INITIAL_FEED_SPEC,
-                 min_n_stages: int = 2, max_n_stages: int = 100, min_pressure: float = 0.1,
-                 max_pressure: float = 10.0,
-                 max_steps: int = 30):
+    def __init__(self,
+                 initial_feed_spec: StreamSpecification = DEFAULT_INITIAL_FEED_SPEC,
+                 product_spec: ProductSpecification = ProductSpecification(purity=0.95),
+                 n_stages_bounds: Tuple[int, int] = (2, 100),
+                 pressure_bounds: Tuple[float, float] = (0.01, 10),
+                 reflux_ratio_bounds: Tuple[float, float] = (0.01, 20.0),
+                 max_steps: int = 30,
```

Comment:
max_steps is the maximum number of columns?

Reply:
Maximum number of environment steps (not separating a stream also counts as an environment step)